src/test/java/org/springframework/data/jpa/repository/query/StringQueryUnitTests.java

```diff
@@ -540,6 +540,34 @@ public void isNotDefaultProjection() {
 		softly.assertAll();
 	}
 
+	@Test // DATAJPA-1652
+	public void usingPipesWithNamedParameter() {
+
+		String queryString = "SELECT u FROM User u WHERE u.lastname LIKE '%'||:name||'%'";
+		StringQuery query = new StringQuery(queryString);
+
+		softly.assertThat(query.getQueryString()).isEqualTo(queryString);
+		softly.assertThat(query.hasParameterBindings()).isTrue();
+		softly.assertThat(query.getParameterBindings()).hasSize(1);
+		softly.assertThat(query.getParameterBindings().get(0).getName()).isEqualTo("name");
+
+		softly.assertAll();
+	}
+
+	@Test // DATAJPA-1652
+	public void usingGreaterThanWithNamedParameter() {
+
+		String queryString = "SELECT u FROM User u WHERE :age<u.age";
```

Comment:
The name says "greaterThan" but the operator used is "less than"

Reply:
:+1: thanks (yeah, traded `>` for `<`, expression looked a bit more comprehensible to me)